working for next Redmine version V4.0.0

Utopism opened the pull request:
Please check this pull request, that makes the plugin working with Redmine V4.0.0.

I have chosen your fork because it seems to be the most active.

Thanks a lot.

Utopism committed:
remvoved upstream, replaced by js chart

https://github.com/redmine/redmine/commit/ad5a4c5d107d12d3c160db6bdd852bbbbcde2178
nouveau fichier : lib/SVG/GPL.txt
nouveau fichier : lib/SVG/Graph/Bar.rb
nouveau fichier : lib/SVG/Graph/BarBase.rb
nouveau fichier : lib/SVG/Graph/BarHorizontal.rb
nouveau fichier : lib/SVG/Graph/Graph.rb
nouveau fichier : lib/SVG/Graph/Line.rb
nouveau fichier : lib/SVG/Graph/Pie.rb
nouveau fichier : lib/SVG/Graph/Plot.rb
nouveau fichier : lib/SVG/Graph/Schedule.rb
nouveau fichier : lib/SVG/Graph/TimeSeries.rb
nouveau fichier : lib/SVG/LICENSE.txt

Utopism committed:
Tested on Redmine 4.0.0 and Ruby 2.3.1

modifié :         README.rdoc

Utopism committed:
ko language header fixed, causes weird translation issue (translations lost)

modifié :         config/locales/ko.yml

Utopism committed:
_list_simple.html.erb removed upstream

nouveau fichier : app/views/issues/_list_simple.html.erb

Utopism committed:
before_filter -> before_action, find(all, ...) replaced

modifié :         app/controllers/graphs_controller.rb

ghost commented:
Thank you, Utopism.
Your Patch is working good with Redmine 4.0.3+Ruby2.5.5(Rails5.2.2.1), after these fixes:
lib/SVG/Graph/Schedule.rb:
`require 'parsedate'` --> `require 'date'`
`ParseDate.parsedate(...` --> `Date.parse(...`

Commit:
Parsedate.parsedate -> Date.parse

thanks n0nel
modifié :         lib/SVG/Graph/Schedule.rb
modifié :         lib/SVG/Graph/TimeSeries.rb

Utopism commented:
Thanks n0nel, I am updating the PR.

Utopism commented:
Works with Redmine V4.0.3

ghost commented:
Works good with **Redmine 4.1.0** & **Ruby 2.6.5**

ashrafalzyoud commented:
```
cannot load such file -- target_version_graph_hook (LoadError)
  /home/redmine/.rbenv/versions/3.1.2/lib/ruby/gems/3.1.0/gems/polyglot-0.3.5/lib/polyglot.rb:65:in `require'
  /home/redmine/.rbenv/versions/3.1.2/lib/ruby/gems/3.1.0/gems/polyglot-0.3.5/lib/polyglot.rb:65:in `require'
  /home/redmine/.rbenv/versions/3.1.2/lib/ruby/gems/3.1.0/gems/zeitwerk-2.6.1/lib/zeitwerk/kernel.rb:35:in `require'
  /home/redmine/.rbenv/versions/3.1.2/lib/ruby/gems/3.1.0/gems/activesupport-6.1.7/lib/active_support/dependencies/zeitwerk_integration.rb:51:in `require_dependency'
  /var/www/5.0.2/plugins/redmine_graphs/init.rb:3:in `<top (required)>'
  /var/www/5.0.2/lib/redmine/plugin_loader.rb:31:in `load'
  /var/www/5.0.2/lib/redmine/plugin_loader.rb:31:in `run_initializer'
  /var/www/5.0.2/lib/redmine/plugin_loader.rb:108:in `each'
  /var/www/5.0.2/lib/redmine/plugin_loader.rb:108:in `block in load'
  /home/redmine/.rbenv/versions/3.1.2/lib/ruby/gems/3.1.0/gems/activesupport-6.1.7/lib/active_support/callbacks.rb:427:in `instance_exec'
  /home/redmine/.rbenv/versions/3.1.2/lib/ruby/gems/3.1.0/gems/activesupport-6.1.7/lib/active_support/callbacks.rb:427:in `block in make_lambda'
  /home/redmine/.rbenv/versions/3.1.2/lib/ruby/gems/3.1.0/gems/activesupport-6.1.7/lib/active_support/callbacks.rb:198:in `block (2 levels) in halting'
  /home/redmine/.rbenv/versions/3.1.2/lib/ruby/gems/3.1.0/gems/activesupport-6.1.7/lib/active_support/callbacks.rb:604:in `block (2 levels) in default_terminator'
  /home/redmine/.rbenv/versions/3.1.2/lib/ruby/gems/3.1.0/gems/activesupport-6.1.7/lib/active_support/callbacks.rb:603:in `catch'
  /home/redmine/.rbenv/versions/3.1.2/lib/ruby/gems/3.1.0/gems/activesupport-6.1.7/lib/active_support/callbacks.rb:603:in `block in default_terminator'
  /home/redmine/.rbenv/versions/3.1.2/lib/ruby/gems/3.1.0/gems/activesupport-6.1.7/lib/active_support/callbacks.rb:199:in `block in halting'
  /home/redmine/.rbenv/versions/3.1.2/lib/ruby/gems/3.1.0/gems/activesupport-6.1.7/lib/active_support/callbacks.rb:512:in `block in invoke_before'
  /home/redmine/.rbenv/versions/3.1.2/lib/ruby/gems/3.1.0/gems/activesupport-6.1.7/lib/active_support/callbacks.rb:512:in `each'
  /home/redmine/.rbenv/versions/3.1.2/lib/ruby/gems/3.1.0/gems/activesupport-6.1.7/lib/active_support/callbacks.rb:512:in `invoke_before'
  /home/redmine/.rbenv/versions/3.1.2/lib/ruby/gems/3.1.0/gems/activesupport-6.1.7/lib/active_support/callbacks.rb:105:in `run_callbacks'
  /home/redmine/.rbenv/versions/3.1.2/lib/ruby/gems/3.1.0/gems/activesupport-6.1.7/lib/active_support/reloader.rb:88:in `prepare!'
  /home/redmine/.rbenv/versions/3.1.2/lib/ruby/gems/3.1.0/gems/railties-6.1.7/lib/rails/application/finisher.rb:124:in `block in <module:Finisher>'
  /home/redmine/.rbenv/versions/3.1.2/lib/ruby/gems/3.1.0/gems/railties-6.1.7/lib/rails/initializable.rb:32:in `instance_exec'
  /home/redmine/.rbenv/versions/3.1.2/lib/ruby/gems/3.1.0/gems/railties-6.1.7/lib/rails/initializable.rb:32:in `run'
  /home/redmine/.rbenv/versions/3.1.2/lib/ruby/gems/3.1.0/gems/railties-6.1.7/lib/rails/initializable.rb:61:in `block in run_initializers'
  /home/redmine/.rbenv/versions/3.1.2/lib/ruby/3.1.0/tsort.rb:228:in `block in tsort_each'
  /home/redmine/.rbenv/versions/3.1.2/lib/ruby/3.1.0/tsort.rb:350:in `block (2 levels) in each_strongly_connected_component'
  /home/redmine/.rbenv/versions/3.1.2/lib/ruby/3.1.0/tsort.rb:431:in `each_strongly_connected_component_from'
  /home/redmine/.rbenv/versions/3.1.2/lib/ruby/3.1.0/tsort.rb:349:in `block in each_strongly_connected_component'
  /home/redmine/.rbenv/versions/3.1.2/lib/ruby/3.1.0/tsort.rb:347:in `each'
  /home/redmine/.rbenv/versions/3.1.2/lib/ruby/3.1.0/tsort.rb:347:in `call'
  /home/redmine/.rbenv/versions/3.1.2/lib/ruby/3.1.0/tsort.rb:347:in `each_strongly_connected_component'
  /home/redmine/.rbenv/versions/3.1.2/lib/ruby/3.1.0/tsort.rb:226:in `tsort_each'
  /home/redmine/.rbenv/versions/3.1.2/lib/ruby/3.1.0/tsort.rb:205:in `tsort_each'
  /home/redmine/.rbenv/versions/3.1.2/lib/ruby/gems/3.1.0/gems/railties-6.1.7/lib/rails/initializable.rb:60:in `run_initializers'
  /home/redmine/.rbenv/versions/3.1.2/lib/ruby/gems/3.1.0/gems/railties-6.1.7/lib/rails/application.rb:391:in `initialize!'
  /var/www/5.0.2/config/environment.rb:16:in `<top (required)>'
  config.ru:3:in `require'
  config.ru:3:in `block in <main>'
  /home/redmine/.rbenv/versions/3.1.2/lib/ruby/gems/3.1.0/gems/rack-2.2.4/lib/rack/builder.rb:125:in `instance_eval'
  /home/redmine/.rbenv/versions/3.1.2/lib/ruby/gems/3.1.0/gems/rack-2.2.4/lib/rack/builder.rb:125:in `initialize'
  config.ru:1:in `new'
  config.ru:1:in `<main>'
  /usr/share/passenger/helper-scripts/rack-preloader.rb:101:in `eval'
  /usr/share/passenger/helper-scripts/rack-preloader.rb:101:in `preload_app'
  /usr/share/passenger/helper-scripts/rack-preloader.rb:189:in `block in <module:App>'
  /usr/lib/ruby/vendor_ruby/phusion_passenger/loader_shared_helpers.rb:397:in `run_block_and_record_step_progress'
  /usr/share/passenger/helper-scripts/rack-preloader.rb:188:in `<module:App>'
  /usr/share/passenger/helper-scripts/rack-preloader.rb:30:in `<module:PhusionPassenger>'
  /usr/share/passenger/helper-scripts/rack-preloader.rb:29:in `<main>'
```

ashrafalzyoud commented:
can u update it in redmine 5
@Utopism 
@Mange 
@nikosd 
@danielpassos
